solidity/nft_token.sol

```diff
@@ -0,0 +1,254 @@
+import "./library/spl_token.sol";
+import "./library/system_instruction.sol";
+
+struct Position {
+    address whirlpool;      // 32
+    address position_mint;  // 32
+    uint128 liquidity;      // 16
+    int32 tick_lower_index; // 4
+    int32 tick_upper_index; // 4
+}
+
+@program_id("DPvSokZKEAYqh6jjmGDUyd4L881uAWg4rFVeA6rec2fr")
+contract nft_token {
+    address public orca;
+    address public whirlpool;
```

Comment:
What's whirlpool?

Reply:
It's an account that holds the overall pool position with the full liquidity.